src/reflect/scala/reflect/internal/util/ThreadLocalStats.java

```diff
@@ -0,0 +1,39 @@
+package scala.reflect.internal.util;
+
+import scala.reflect.internal.util.Statistics;
+
+public class ThreadLocalStats {
+    public static final ThreadLocal<Statistics> statisticsLocal = new ThreadLocal<Statistics>();
+
+    // We can't use `Statistics.Counter` because of https://github.com/scala/bug/issues/10490
+    public static final ThreadLocal<Object> fileExistsCountLocal = new ThreadLocal<Object>();
+    public static final ThreadLocal<Object> fileIsFileCountLocal = new ThreadLocal<Object>();
+    public static final ThreadLocal<Object> fileIsDirectoryCountLocal = new ThreadLocal<Object>();
+
+    // Cache statistics when they are read from thread local for the first time
+    public static Statistics cachedStatistics = null;
```

Comment:
That seems to cause races when there are multiple compilers running in parallel.

Reply:
Good point, didn't realise this is the case. I'll think of a way to fix this.

Reply:
I may need to remove `IOStats` altogether for this to be friendly to multiple compilers running in paralle.

Reply:
I would start by removing the cache here and see if there's a performance problem going through the `ThreadLocal`. Compared to IO, accessing a thread local is probably inexpensive.

Reply:
I would advise to just remove these counters altogether. I think I added them to demonstrate that some optimizations in classpath scanning reduced the IO operations. It's easy enough to temporarily add a global counter for such things during future optimizations in this area.

Reply:
Okay, I'll remove them then.